packages/react-components/react-avatar/src/components/AvatarGroupOverflow/useAvatarGroupOverflow.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { AvatarGroupContext } from '../../contexts/AvatarGroupContext';
+import { defaultAvatarGroupSize } from '../AvatarGroup/useAvatarGroup';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import { MoreHorizontalRegular } from '@fluentui/react-icons';
+import { OnOpenChangeData, OpenPopoverEvents, Popover, PopoverSurface } from '@fluentui/react-popover';
+import { useContextSelector } from '@fluentui/react-context-selector';
+import type { AvatarGroupOverflowProps, AvatarGroupOverflowState } from './AvatarGroupOverflow.types';
+import { Tooltip } from '@fluentui/react-tooltip';
+
+/**
+ * Create the state required to render AvatarGroupOverflow.
+ *
+ * The returned state can be modified with hooks such as useAvatarGroupOverflowStyles_unstable,
+ * before being passed to renderAvatarGroupOverflow_unstable.
+ *
+ * @param props - props from this instance of AvatarGroupOverflow
+ * @param ref - reference to root HTMLElement of AvatarGroupOverflow
+ */
+export const useAvatarGroupOverflow_unstable = (props: AvatarGroupOverflowProps): AvatarGroupOverflowState => {
+  const size = useContextSelector(AvatarGroupContext, ctx => ctx.size) ?? defaultAvatarGroupSize;
+  const layout = useContextSelector(AvatarGroupContext, ctx => ctx.layout);
+  const [isPopoverOpen, setIsPopoverOpen] = React.useState(false);
+  const {
+    indicator = size < 24 ? 'icon' : 'count',
+    count = React.Children.count(props.children),
+    children,
+    ...restOfProps
+  } = props;
+  let triggerButtonChildren;
+
+  const handleOnPopoverChange = (e: OpenPopoverEvents, data: OnOpenChangeData) => {
+    setIsPopoverOpen(data.open);
+  };
```

Comment:
Need to call the user-provided callback here:
```suggestion
  const handleOnPopoverChange = (e: OpenPopoverEvents, data: OnOpenChangeData) => {
    restOfProps.onOpenChange?.(e, data);
    setIsPopoverOpen(data.open);
  };
```